src/parser/jobs/mnk/modules/Steppies.tsx

```diff
@@ -0,0 +1,122 @@
+import {Plural, Trans} from '@lingui/react'
+import React from 'react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+
+import {CastEvent, DamageEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import Combatants from 'parser/core/modules/Combatants'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+const LEAD_BOOT_POTENCY = 300
+
+// Typically a player with lag can derp positional in opener, this usually happens 2 times per fight.
+const CRIT_BOOT_SEVERITY = {
+	1: SEVERITY.MEDIUM,
+	2: SEVERITY.MAJOR,
+}
+
+// 3 is pretty much "you ruined a Perfect Balance you turkey".
+const WEAK_BOOT_SEVERITY = {
+	1: SEVERITY.MEDIUM,
+	3: SEVERITY.MAJOR,
+}
+
+class Boot {
+	crit: boolean
+	timestamp: number
+	weak: boolean
+
+	constructor(weak: boolean, timestamp: number) {
+		this.crit = false
+		this.weak = weak
+		this.timestamp = timestamp
+	}
+}
+
+export default class Steppies extends Module {
+	static handle = 'steppies'
+
+	@dependency private checklist!: Checklist
+	@dependency private combatants!: Combatants
+	@dependency private suggestions!: Suggestions
+
+	private currentBoot?: Boot
+	private steppies: Boot[] = []
+
+	protected init(): void {
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.BOOTSHINE.id}, this.onCast)
+		this.addHook('damage', {by: 'player', abilityId: ACTIONS.BOOTSHINE.id}, this.onStep)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent): void {
+		this.currentBoot = new Boot(this.combatants.selected.hasStatus(STATUSES.LEADEN_FIST.id), event.timestamp)
```

Comment:
Any particular reason to have this in its own separate hook instead of just creating the `Boot` object in the `damage` hook?

Reply:
We need to know buff state when the cast happens, but then hit state on damage.

Reply:
As discussed in Discord.